### PR TITLE
fix(scan): accumulator passes current index

### LINKED ISSUE #1614: Why no index or observable arguments in scan's accumulator?
Sorry if this kind of question is out of place or out of format.

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

This PR adds small changes to `accumulator` of `scan` to pass index of source. Still does not passes source observable.

**Related issue (if exists):**

closes #1614

### CODE REVIEW

File: src/operator/scan.ts [hunk 1]
```diff
@@ -32,23 +32,23 @@ import {Subscriber} from '../Subscriber';
  * @see {@link mergeScan}
  * @see {@link reduce}
  *
- * @param {function(acc: R, value: T): R} accumulator The accumulator function
+ * @param {function(acc: R, value: T, , index: number): R} accumulator The accumulator function
```
Comment (staltz) [inline src/operator/scan.ts]: @kwonoj Can you fix these commas? Should be `, ` not `, , `. Also limiting the line width to 80 chars would be good.

**Comment by staltz** [comment]
Besides the comment I placed in code, looks good to merge in my opinion.

**Comment by kwonoj** [comment]
I'll check this in later today if there isn't other suggestions around.

**Comment by kwonoj** [comment]
Merged with a3ec896.

**Pull request closed without merging**